### PR TITLE
Updates for SR2024

### LINKED ISSUE #40: Dark theme
It would be really nice to have a dark theme, preferable that works with the system UI preference.

### PR DESCRIPTION
**Pull request by RealOrangeOne**
Closes #59
Closes #61
Fixes #40 
Includes #50 and #51

Changes are currently deployed and seem to be working.

### CODE REVIEW

**Comment by PeterJCLaw** [comment]
> Includes #50 and #51

Suggest we get those separately merged first?

**Comment by RealOrangeOne** [comment]
We could, but I specifically wanted to test these changes all together. I've reviewed and approved those separately, so I think it's ok to have them merged as part of this.

**Review by trickeydan — APPROVED** [review]
I think this works as a minimum. I might find some time to give this project a bit of TLC before the competition.

File: requirements-dev.txt [hunk 1]
```diff
@@ -154,15 +155,6 @@ sqlparse==0.4.4
     # via
     #   -r requirements.txt
     #   django
-tomli==2.0.1
```
Comment (PeterJCLaw) [inline requirements-dev.txt]: Dropping `tomli` and other changes here seem out of step with what happens if you run the requirements compiling script. Is that expected?

**Review by PeterJCLaw — APPROVED** [review]
Changes seem fine. I've only looked at the first set of changes from @RealOrangeOne and haven't re-reviewed the content from #63. Also haven't actually run this locally -- this review is from code inspection alone.

File: requirements-dev.txt (hunk 1 above)
Comment (RealOrangeOne) [reply requirements-dev.txt]: It happens if `pip-tools` is run on Python 3.11+, as `tomli` isn't needed then. This change is no longer in the PR.

Comment (PeterJCLaw) [reply requirements-dev.txt]: :thumbsup: I suspected it might be that. #65 aims to help with that sort of thing.

**Review by trickeydan — APPROVED** [review]
I think this is good for a first pass. Let's merge and deploy this and if we make any other improvements we can consider those individually. 

I had a go at implementing markdown / rich text earlier but ran out of time. I remember we really wished we had bullet points and quote last year.

**Review by raccube — APPROVED** [review]

**Pull request merged**